equinox/nn/_embedding.py

```diff
@@ -162,6 +162,11 @@ def process_heads(
 
     embedding_size: int = field(static=True)
     theta: float = field(static=True, default=10_000.0)
+    dtype: Any = field(static=True, default=None)
+
+    def __post_init__(self):
+        if self.dtype is None:
+            self.dtype = default_floating_dtype()
```

Comment:
I think this could be `dtype: Any = field(static=True, default_factory=default_floating_dtype)`.

Reply:
You are right.... I didn't have a default factory in my mental model, but according to docs it is indeed equivalent to my code: 

_If a field is excluded from [__init__()](https://docs.python.org/3/reference/datamodel.html#object.__init__) (using init=False) and the field also specifies default_factory, then the default factory function will always be called from the generated __init__() function. This happens because there is no other way to give the field an initial value._

Anyhow, thanks!